### PR TITLE
stm32f7: add DCMI registers

### PR DESCRIPTION
**Pull request by smolin35**
This patch adds definition of the DCMI registers for sm32f7 MCU series. 
I tested part of this code with nucleo-f746zg and nucleo-f767zi boards to get data from an external ADC.

### CODE REVIEW

File: include/libopencm3/stm32/f7/dcmi.h [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+ *
+ * @version 1.0.0
+ *
+ * @date 2019-09-04
+ *
+ * LGPL License Terms @ref lgpl_license
+ */
+
+
+//TODO check
```
Comment (karlp) [inline include/libopencm3/stm32/f7/dcmi.h:15]: check what? if this is finished, remove this please

**Comment by karlp** [comment]
As far as I can see, this is 5 extra bits in the CR register?  This needs to be pulled out to a common dcmi peripheral header, with just the special extras for f7, not just copied over please.

**Comment by smolin35** [comment]
Sorry for the long answer.

"As far as I can see, this is 5 extra bits in the CR register? "
Yes it is.
If I understand you right, you would like I put the common part for f4 and f7  in include/libopencm3/stm32/common/dcmi_common_f47.h, and leave the specific extras for f7 and f4 in include/libopencm3/stm32/fx/dcmi. h?

**Comment by karlp** [comment]
This looks good, but the doxygen is now broken, f4 has no docs, and f7 only has the specific docs.

**Comment by karlp** [comment]
NVM, I figured out how to fix the doxygen! :)

**Comment by karlp** [comment]
this has now landed in master thanks!

**Pull request closed without merging**